### ISSUE TITLE
Exif too large

### ISSUE THREAD

**Issue by fenos** [imgproxy/imgproxy#1263]
Hi,
Imgproxy is throwing a 500 error with the message: `exif too large` when trying to resize certain images.

Version: 3.21.0

here is the stacktrace:

```
"stack":"/app/vips/vips.go:401 github.com/imgproxy/imgproxy/v3/vips.(*Image).Save\n/app/processing/processing.go:339
github.com/imgproxy/imgproxy/v3/processing.ProcessImage\n/app/processing_handler.go:428 
main.handleProcessing.func3\n/app/processing_handler.go:426 main.handleProcessing\n/app/server.go:114 
main.buildRouter.withCORS.func1\n/app/server.go:165 main.buildRouter.withPanicHandler.func2\n/app/server.go:103 
main.buildRouter.withMetrics.func3\n/app/router/router.go:102 github.com/imgproxy/imgproxy/v3/router.
(*Router).ServeHTTP\n/usr/local/go/src/net/http/server.go:2939 net/http.serverHandler.ServeHTTP\n/usr/local/go/src/net/http/server.go:2010 net/http.(*conn).serve","status":500
```

I've used `exiftool` to provide more information about the image causing this issue:

```
ExifTool Version Number         : 12.76
File Name                       : exif-too-large.png
Directory                       : /Users/fenos/Documents
File Size                       : 5.5 MB
File Modification Date/Time     : 2024:03:30 13:36:20+00:00
File Access Date/Time           : 2024:03:30 13:36:48+00:00
File Inode Change Date/Time     : 2024:03:30 13:36:26+00:00
File Permissions                : -rw-r--r--
File Type                       : PNG
File Type Extension             : png
MIME Type                       : image/png
Image Width                     : 1024
Image Height                    : 1024
Bit Depth                       : 8
Color Type                      : RGB
Compression                     : Deflate/Inflate
Filter                          : Adaptive
Interlace                       : Noninterlaced
JUMD Type                       : (c2pa)-0011-0010-800000aa00389b71
JUMD Label                      : c2pa
Title                           : 46b0ae4dd02d9da14276a6434acbbf44c45ef19bee89746aa7cd86bb3c6c4ee5.png
Format                          : application/octet-stream
Instance ID                     : xmp:iid:3fa103c5-c398-4219-8484-fec1f55c5c23
Relationship                    : parentOf
Stds Schema-org Creative Work Salt: 03edad1d22fcb76784b68a492c0e938a
Context                         : https://schema.org
Type                            : CreativeWork
Date Published                  : 2023-05-10T23:02:36+00:00
Publisher Type                  : Organization
Publisher Name                  : Stability AI
Com Adobe Generative-ai Salt    : 34b4d8bd08a12528e21464cec324b3b1
Description                     : stable-diffusion-xl-1024
Version                         : 1.0
Exclusions Start                : 33
Exclusions Length               : 16944
Name                            : jumbf manifest
Alg                             : sha256
Hash                            : (Binary data 32 bytes, use -b option to extract)
Pad                             : (Binary data 13 bytes, use -b option to extract)
Claim generator                 : Stability_ContentCredentials_Service/v0.1 c2pa-rs/0.31.0
Claim generator info            : null
Signature                       : self#jumbf=c2pa.signature
Assertions Url                  : self#jumbf=c2pa.assertions/c2pa.ingredient, self#jumbf=c2pa.assertions/stds.schema-org.CreativeWork, self#jumbf=c2pa.assertions/com.adobe.generative-ai, self#jumbf=c2pa.assertions/c2pa.hash.data
Assertions Hash                 : (Binary data 32 bytes, use -b option to extract), (Binary data 32 bytes, use -b option to extract), (Binary data 32 bytes, use -b option to extract), (Binary data 32 bytes, use -b option to extract)
Item 0                          : (Binary data 2664 bytes, use -b option to extract)
Item 1 Pad                      : (Binary data 12298 bytes, use -b option to extract)
Item 2                          : null
Item 3                          : (Binary data 256 bytes, use -b option to extract)
Exif Byte Order                 : Big-endian (Motorola, MM)
Warning                         : IFD0 pointer references previous JUMBFBox directory
Image Size                      : 1024x1024
Megapixels                      : 1.0
```

The metadata seems standard, not entirely sure why Vips is throwing this error, any idea?
Thanks

**Comment by DarthSim** [comment]
Hey @fenos!

Can I take a look at the image?

**Comment by fenos** [comment]
I've DM you on Slack with the image causing the issue 👍

**Comment by fenos** [comment]
when running `vipsheader -a` i get the following output:

```
vipsheader -a ~/Documents/exif-too-large.png

(vipsheader:24060): VIPS-WARNING **: 18:31:32.977: image_set: bad exif data
/Users/fenos/Documents/exif-too-large.png: 1024x1024 uchar, 3 bands, srgb, pngload
width: 1024
height: 1024
bands: 3
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 2.83465
yres: 2.83465
filename: /Users/fenos/Documents/exif-too-large.png
vips-loader: pngload
exif-data: 3953601 bytes of binary data
bits-per-sample: 8
```

which seems that the exif-data is ~4MB this should be the problem. 
I'm wondering if we should handle this gracefully or returning a more friendly error if the exif-data is too large

**Comment by DarthSim** [comment]
I made a PR to libvips that increases the EXIF size limit: https://github.com/libvips/libvips/pull/3917

**Comment by DarthSim** [comment]
EXIF size limit is increased to 8MB in the `latest` build.

**Comment by DarthSim** [comment]
The fix is released in `v3.24.0`
